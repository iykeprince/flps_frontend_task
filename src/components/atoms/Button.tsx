import React from "react";

export interface IButton {
  label: string;
  showIconRight: boolean;
  onClick?: any;
}

const Button = ({ label, showIconRight = false, onClick }: IButton) => (
  <button className="flex items-center" onClick={onClick}>
    {label} {showIconRight && <i className="fa fa-chevron-right ml-2"></i>}
  </button>
);

export default Button;
