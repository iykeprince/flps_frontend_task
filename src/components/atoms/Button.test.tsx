import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import Button, { IButton } from "./Button";

describe("<Button />", () => {
  let component: ShallowWrapper;
  let mockCallBack = jest.fn();
  beforeEach(() => {
    component = shallow(
      <Button label="Button" showIconRight={false} onClick={mockCallBack} />
    );
  });
  test("Button exists", () => {
    expect(component.length).toBe(1);
  });
  test("it should render our snapshot correctly", () => {
    expect(component).toMatchSnapshot();
  });
  test("it should react when clicked", () => {
    component.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
