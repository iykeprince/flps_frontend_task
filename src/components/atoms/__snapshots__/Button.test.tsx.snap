// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> it should render our snapshot correctly 1`] = `
<button
  className="flex items-center"
  onClick={[MockFunction]}
>
  Button
   
</button>
`;
